components/plugin_marketplace/marketplace_item/marketplace_item_app/marketplace_item_app.tsx

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {FormattedMessage} from 'react-intl';
+
+import type {MarketplaceLabel} from 'mattermost-redux/types/marketplace';
+
+import MarketplaceItem from '../marketplace_item';
+
+export type MarketplaceItemAppProps = {
+    id: string;
+    name: string;
+    description?: string;
+    homepage_url?: string;
+    root_url: string;
+
+    installed: boolean;
+    labels?: MarketplaceLabel[];
+
+    trackEvent: (category: string, event: string, props?: unknown) => void;
+
+    actions: {
+        installApp: (id: string, url: string) => void;
+        closeMarketplaceModal: () => void;
+    };
+};
+
+export default class MarketplaceItemApp extends React.PureComponent <MarketplaceItemAppProps> {
+    onInstall = (): void => {
+        this.props.trackEvent('plugins', 'ui_marketplace_install_app', {
+            app_id: this.props.id,
+        });
+
+        this.props.actions.installApp(this.props.id, this.props.root_url);
+
+        this.props.actions.closeMarketplaceModal();
```

Comment:
Just to check - is this what the design says - to close the modal once the install button is pressed?

Reply:
No. The new design implement the whole installation flow inside the Marketplace Modal. Closing the modal is a compromise until the new design is implemented.

Reply:
Thanks, let's make sure there is a ticket opened for the new design then.